beacon-chain/monitor/process_block.go

```diff
@@ -41,12 +42,12 @@ func (s *Service) processBlock(ctx context.Context, b block.SignedBeaconBlock) {
 	}
 
 	currEpoch := slots.ToEpoch(blk.Slot())
-	s.monitorLock.RLock()
+	s.RLock()
 	lastSyncedEpoch := s.lastSyncedEpoch
-	s.monitorLock.RUnlock()
+	s.RUnlock()
 
 	if currEpoch != lastSyncedEpoch &&
-		slots.SyncCommitteePeriod(currEpoch) == slots.SyncCommitteePeriod(s.lastSyncedEpoch) {
+		slots.SyncCommitteePeriod(currEpoch) == slots.SyncCommitteePeriod(lastSyncedEpoch) {
```

Comment:
This was a bug. `s.lastSyncedEpoch` was missing a lock. To make it simpler, I just used `lastSyncedEpoch` from above. Ok?

Reply:
Thanks! this was a bug added when I added the lock!